easy-random-core/src/main/java/org/jeasy/random/RecordFactory.java

```diff
@@ -67,30 +67,26 @@ private <T> T createRandomRecord(Class<T> recordType, RandomizationContext conte
         Field[] fields = recordType.getDeclaredFields();
         RecordComponent[] recordComponents = recordType.getRecordComponents();
         Object[] randomValues = new Object[recordComponents.length];
-        if (context.hasExceededRandomizationDepth()) {
-            for (int i = 0; i < recordComponents.length; i++) {
-                Class<?> type = recordComponents[i].getType();
-                randomValues[i] = DepthLimitationObjectFactory.produceEmptyValueForField(type);
-            }
-        } else {
-            for (int i = 0; i < recordComponents.length; i++) {
-                context.pushStackItem(new RandomizationContextStackItem(recordType, null));
-                Class<?> type = recordComponents[i].getType();
-                try {
-                    if (isRecord(type)) {
-                        randomValues[i] = easyRandom.doPopulateBean(type, context);
+
+        for (int i = 0; i < recordComponents.length; i++) {
+            context.pushStackItem(new RandomizationContextStackItem(recordType, null));
+            Class<?> type = recordComponents[i].getType();
+            try {
+                if (isRecord(type)) {
+                    if (context.hasExceededRandomizationDepth()) {
```

Comment:
Can we move this check outside of this if condition?

Reply:
`recordFieldPopulator.populateField` already had the logic, but as you pointed out it can be outside of this if the condition and then it is possible to remove it from `recordFieldPopulator.populateField` so that the `hasExceededRandomizationDepth` is checked only in one place for records.

Reply:
@dvgaba how about merging the change?

Reply:
@dvgba, could you let me know when I can expect a decision regarding the merging of this PR? Your response would be greatly appreciated.

Reply:
My perspective might be biased, but I would recommend making it the default approach. For the majority of cases, this choice might not have a significant impact since reaching the recursion limit without recursive data structures is rare. However, when working with such data, the prevailing guideline for production code is to employ empty collections rather than nulls. Production code factories create initialized collections, and algorithms — tested with data generated by easy-random — typically assume the absence of nulls. Therefore, I believe that in scenarios where this distinction matters, initializing the collections is more convenient.